app/src/main/java/zegreanu/cristi/geoquiz/CheatActivity.kt

```diff
@@ -8,16 +8,23 @@ import android.widget.TextView
 import androidx.appcompat.app.AppCompatActivity
 
 class CheatActivity : AppCompatActivity() {
+    private val KEY_ANSWER_SHOWN = "answer_shown"
+
     private lateinit var showAnswerButton: Button
     private lateinit var answerTextView: TextView
 
+    private var isAnswerShown: Boolean = false
     private var answerIsTrue: Boolean = false
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_cheat)
 
         answerIsTrue = intent.getBooleanExtra(EXTRA_ANSWER_IS_TRUE, false)
+        savedInstanceState?.let {
+            isAnswerShown = it.getBoolean(KEY_ANSWER_SHOWN, false)
```

Comment:
Same, to avoid duplicating the default value, you can do `isAnswerShown = it.getBoolean(KEY_ANSWER_SHOWN, isAnswerShown)`